codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/RpcWorker.res

```diff
@@ -90,9 +90,19 @@ module Make = (
     if fromBlock > currentBlockHeight {
       let nextBlock = await waitForBlockGreaterThanCurrentHeight(~currentBlockHeight, ~logger)
 
-      setCurrentBlockHeight(nextBlock)
+      let currentBlockHeight = if nextBlock > currentBlockHeight {
+        setCurrentBlockHeight(nextBlock)
+        nextBlock
+      } else {
+        currentBlockHeight
+      }
```

Comment:
My assumption for the bug occurring is that the block listener provided by ethers can somehow return a block lower than what we already have stored. So I'm adding a check here and recursing through the function to ensure we never query a fromBlock above the current height